src/libsystemd/sd-id128/id128-util.c

```diff
@@ -102,14 +109,15 @@ int id128_read_fd(int fd, Id128FormatFlag f, sd_id128_t *ret) {
         return r == -EINVAL ? -EUCLEAN : r;
 }
 
-int id128_read(const char *root, const char *p, Id128FormatFlag f, sd_id128_t *ret) {
+int id128_read(const char *root, const char *path, Id128FormatFlag f, sd_id128_t *ret) {
```

Comment:
Now we have this weird inconsistency where `id128_read()` works on a root directory and chases symlinks and `id128_read_at()` works on a directory file descriptor and does not chase symlinks.

I would make this consistent by only having `id128_read()` take a path and not a root directory. and then have `id128_get_machine_id()` and `id128_get_machine_id_at()` which operate on a root directory so that the responsibilities are nicely separated.